compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -598,36 +603,80 @@ private BType getSmallestBuiltInUnsignedIntSubTypeContainingTypes(BType lhsType,
     }
 
     void generatePlatformIns(JInstruction ins, int localVarOffset) {
-        if (ins.jKind == JInsKind.JCAST) {
-            JCast castIns = (JCast) ins;
-            BType targetType = castIns.targetType;
-            this.loadVar(castIns.rhsOp.variableDcl);
-            jvmCastGen.generatePlatformCheckCast(this.mv, this.indexMap, castIns.rhsOp.variableDcl.type, targetType);
-            this.storeToVar(castIns.lhsOp.variableDcl);
-        } else if (ins.jKind == JInsKind.CALL) {
-            JMethodCallInstruction callIns = (JMethodCallInstruction) ins;
-            boolean isInterface = callIns.invocationType == INVOKEINTERFACE;
-            int argIndex = 0;
-            String jMethodVMSig = callIns.jMethodVMSig;
-            boolean hasBalEnvParam = jMethodVMSig.startsWith(BAL_ENV_PARAM);
-            if (hasBalEnvParam) {
-                mv.visitTypeInsn(NEW, BAL_ENV_CLASS);
-                mv.visitInsn(DUP);
-                // load the strand
+        switch (ins.jKind) {
```

Comment:
Shall we move some cases (like `LARGE_ARRAY`) to separate functions? To improve readability